jax/_src/lax/parallel.py

```diff
@@ -725,10 +725,14 @@ def _allreduce_abstract_eval(*args, axes, axis_index_groups):
           for arg, named_shape in zip(args, named_shapes)]
 
 def _allreduce_lowering(prim, pos_fn, ctx, *args, axes, axis_index_groups):
-  if axis_index_groups is not None and ctx.module_context.platform == "tpu":
+  if ctx.module_context.lowering_parameters.is_multi_platform:
```

Comment:
Why is the fix here not to pass `platform` explicitly to the lowering rule, as we usually do?

Reply:
That would work too, but it would have the effect of making the all_reduce lowering be platform specific, which would result in multi-platform lowering generating a conditional with identical and fairly large branches. The only thing that is platform specific here is the generation of an error, the actual code generated is the same. This is why I chose this implementation.

Reply:
Discussed offline: a future change will remove `ctx.module_context.platform` and will introduce `ctx.module_context.platforms`, in which case the change here will be simplified, e.g.,

```
if axis_index_groups is not None and "tpu" in ctx.module_context.platforms:
```

Reply:
Update the PR?

Reply:
Added a TODO